app/Http/Controllers/ShopController.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ShopPostRequest;
+use App\Models\Shop;
+use Illuminate\Http\Request;
+
+class ShopController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function index()
+    {
+        $shops = Shop::with('ads')->get();
+        return view('theme.include.shops',compact('shops'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\Shop  $shop
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function show($id)
+    {
+        $shop = Shop::with('products')->whereId($id)->get();
+        if($shop){
+            return view('theme.include.shop-page',compact('shop'));
+        }else{
+            abort(404);
```

Comment:
Burada yapmak istediğin sanırım shop yok ise 404'e göndermek. Böylece url'den değiştirildiğinde shop olmayınca hata vericek. Bunu daha kolay bir şekilde yapabilrisin. 49. satıra show(Shop $shop) şeklinde modeli kullanarak kontrol ettirebilirsin. Eğer shop var ise direkt veritabanından çekip $shop değişkeni olarak sana vericek yok ise kendi 404 e gidicek. Aynısı update,detele,edit fonksiyonların içinde geçerli.

Reply:
Bu kısımı tam anlayamadım. Shop $shop, $id şeklinde mi parametreler alacak ?

Reply:
hayır, show($id) yerine show(Shop $shop) yazıcaksın ve $shop ile işlemlerine devam ediceksin. Değişkenin ismi değişecek çünkü bir id değil artık. Bu fonksiyona bir id geliyor Shop modeli ile veritabanından o id'ye sahip satır komple $shop değişkenine atanıyor.

Kısaca Fonksiyonun içerisinde yazdığın 51. satırda $shop değişkenini veritabanından çekme işlemi ile 52-55. satırları arasında yaptığın kontrol işlemini tek satırda yapmış oluyoruz.

Bunu test edebilirsin fonksiyonunun içerisine "return $shop" at ve bak ne geliyor diye.

Aşağı yukarı şöyle görünmesi lazım fonksyonun;
show(Shop $shop){
    return(route.... ,compact('shop'));
}

Reply:
Bu kısımda (Shop $shop) şeklinde kullanım yapınca $shop üzerinden yaptığım işlemler çalışmadı, örneğin $shop->delete(); diyorum, hata vermiyor, success mesajına da giriyor ama veritabanından silme işlemini yapmıyor. Bu yüzden $id ile ilerleyip Shop::whereId($id)->delete(); şeklinde gitmeye devam edeyim diyorum.

Reply:
(Shop $shop) yaptığın zaman web.php deki route'undaki tanımladığın değişkenin isminide düzenlemen gerekiyor

Örnek:Route::delete('admin/magazalar/{shop}', [ShopController::class,'destroy'])->name('adminShopDelete'); <- buradaki süslü parantez içerisinde değişken tanımlıyorsun fonksiyonda da o değişkeni çekip kullanıyorsun, değişken adları aynı olmak zorunda. Burada id kalmışsa çalışmamış olabilir.